intel_extension_for_transformers/llm/runtime/graph/__init__.py

```diff
@@ -61,44 +61,61 @@ def __import_package(self, model_name):
             raise TypeError("Unspported model type {}!".format(model_name))
         self.module = cpp_model
 
-    def init(self, model_name, **kwargs):
-        config = AutoConfig.from_pretrained(model_name, trust_remote_code=True)
-        model_type = model_maps.get(config.model_type, config.model_type)
-        if model_type == "chatglm" and "chatglm2" in config._name_or_path:
+    def init(self, model_name, not_quant=False, use_cache=False, **quant_kwargs):
+        self.config = AutoConfig.from_pretrained(model_name, trust_remote_code=True)
+        self.tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)
+        model_type = model_maps.get(self.config.model_type, self.config.model_type)
+        if model_type == "chatglm" and "chatglm2" in self.config._name_or_path:
             model_type = "chatglm2"
         self.__import_package(model_type)
 
-        # 1. convert model
-        fp32_bin = "ne_{}_f32.bin".format(model_type)
+        # check cache and quantization
+        output_path = "runtime_outs"
+        if not os.path.exists(output_path):
+            os.makedirs(output_path)
+        fp32_bin = "{}/ne_{}_f32.bin".format(output_path, model_type)
+        quant_bin = "{}/ne_{}_q.bin".format(output_path, model_type)
+
+        if not_quant:
+            self.bin_file = fp32_bin
+        else:
+            self.bin_file = quant_bin
+        if use_cache and os.path.exists(self.bin_file):
+            return
+
         convert_model(model_name, fp32_bin, "f32")
         assert os.path.exists(fp32_bin), "Fail to convert pytorch model"
 
-        # 2. quant model
-        quant_bin = "ne_{}_q.bin".format(model_type)
-        self.module.Model.quant_model(model_path = fp32_bin, out_path = quant_bin, **kwargs)
+        if not_quant:
+            return
+        self.module.Model.quant_model(model_path = fp32_bin, out_path = quant_bin, **quant_kwargs)
         assert os.path.exists(quant_bin), "Fail to quantize model"
         
-        self.model_type = model_type
-        self.bin_file = quant_bin
-        
         # clean
         os.remove(fp32_bin)
 
-    def init_from_bin(self, model_name, model_path, **kwargs):
+    def init_from_bin(self, model_name, model_path, **generate_kwargs):
         self.__import_package(model_name)
         self.model = self.module.Model()
-        self.model.init_model(model_path, **kwargs)
+        if "threads" not in generate_kwargs:
+            threads = os.getenv("OMP_NUM_THREADS")
+            if threads is None:
+                generate_kwargs["threads"] = os.cpu_count() // 2
```

Comment:
why use half cpu for thread num,
assuming hybrid thread is enabling ?

Reply:
update 0fdb9e848fd214fb2770650ec629c049aad1686a,
Use `len(os.sched_getaffinity(0))` to get cores under numa settings.  thanks @DDEle